css/css-contain/contain-layout-baseline-001.html

```diff
@@ -0,0 +1,31 @@
+<!doctype html>
+<html lang=en>
+  <meta charset=utf-8>
+  <title>CSS-contain test: layout containment and baselines</title>
+  <link rel="author" title="Florian Rivoal" href="https://florian.rivoal.net">
+  <meta name=assert content="With contain:layout, for the purpose of the vertical-align property, the containing element is treated as having no baseline.">
+  <link rel="match" href="../reference/ref-filled-green-100px-square.xht">
+  <link rel=help href="https://drafts.csswg.org/css-contain-1/#containment-layout">
+  <meta name="flags" content="">
+
+<style>
+#red {
+  position: absolute;
+  background: red;
+  width: 100px;
+  height: 100px;
+  z-index: -1;
+}
+div {
+  display: inline-block;
+  height: 100px;
+  background: green;
+  width: 50px;
+  contain: layout;
+  color: transparent;
+}
+</style>
+
+<p>Test passes if there is a filled green square and <strong>no red</strong>.</p>
+<aside id=red></div>
+<section><div id=d1></div><div id=d2>a</div></section>
```

Comment:
Then here you won't need the `section` thing, neither the ids for the div elements (they're not used) and you could use `class="green"` for these divs.